Admin manage subjects style

pswk1 opened the pull request:
- updates to styling of manage course topics tab in admin portal to match Figma specs

pswk1 committed:
refactor(manage course topics): rename UI to topics

- rename instances of "tag" to "topic" in manage course topics tab in admin portal

pswk1 committed:
style(manage course topic): modal style

- update styling of create topic modal to match Figma

pswk1 committed:
style(manage course topics): cell widths

- update styling to match normal and edit mode table cell widths

pswk1 committed:
Apply formatting changes

pswk1 committed:
refactor(manage course topics): style updates

- remove unused classes/code
- further update to styling to match Figma specs

pswk1 committed:
Merge branch 'admin-manage-subjects-style' of github.com:omou-org/front-end into admin-manage-subjects-style

pswk1 committed:
Apply formatting changes

pswk1 committed:
refactor (manage course topic): style

- update cell width

pswk1 committed:
Merge branch 'admin-manage-subjects-style' of github.com:omou-org/front-end into admin-manage-subjects-style

pswk1 committed:
Apply formatting changes

pswk1 committed:
refactor(manage course topics): style fix

- update font style of edit mode

pswk1 committed:
Merge branch 'admin-manage-subjects-style' of github.com:omou-org/front-end into admin-manage-subjects-style

pswk1 committed:
Apply formatting changes

pswk1 committed:
fix(manage course topics): update cache after editing

- cache update after editing/updating a topic

pswk1 committed:
Merge branch 'admin-manage-subjects-style' of github.com:omou-org/front-end into admin-manage-subjects-style

pswk1 committed:
Apply formatting changes

dhuangg commented on src/components/FeatureViews/AdminPortal/CreateTopicModal.js:
```diff
@@ -90,57 +95,44 @@ const GET_COURSE_TAGS = gql`
+            const cachedCategory = cache.readQuery({
+                query: GET_COURSE_TAGS,
+            }).courseCategories;
+            const updatedCache = [...cachedCategory, newCategory];
```
Few things:
- Mutations are used for both creating **new** course topics and **updating** existing course topics
- What this means is that this "update" callback function (starting line 108) gets triggered after every creation or update of course topics
- In this function we need to handle both situations (creating a new course topic, or updating an existing course topic)
- Currently the code (line 113) is handling creating new course topics (adding the new category to the end of the list)
- We need to handle the situation when we're updating an existing course topic. To do this, we need to 1) identify where in the course topics (cachedCategory) list is the existing course topic 2) update the course topic there. For example, we want to update the "Math" course topic, so we need to 1) identify where in the list it is (get the index of the "Math" course topic) which we'll say is at index 2. 2) update the course topic at index 2

Your todo will be to implement the logic in the last bullet point

pswk1 replied:
CreateTopicModal.js handles the logic for just creating a new topic (since we're not editing topics inside the modal, right?).
The logic that handles updating the topic at its index is contained in `[ManageCourseTopics](https://github.com/omou-org/front-end/pull/778/commits/d4a3403db8b80ccc0c602b0f0ff493247fdd815e)`, starting at line 139.

pswk1 replied:
CreateTopicModal handles the logic of just creating a new topic (since we're not editing topics within this modal, right?)
The logic that implements updating the topic at its current index is contained in ManageCourseTopics.js starting at line 139

dhuangg replied:
ahh gotcha. thanks!

dhuangg commented on src/components/FeatureViews/AdminPortal/CreateTopicModal.js:
```diff
@@ -90,57 +95,44 @@ const GET_COURSE_TAGS = gql`
-        //         });
-        // }
+        update: (cache, { data }) => {
+            const newCategory = data.createCourseCategory.courseCategory;
```
refactor to "updatedCourseTopic"

dhuangg commented on src/components/FeatureViews/AdminPortal/CreateTopicModal.js:
```diff
@@ -90,57 +95,44 @@ const GET_COURSE_TAGS = gql`
-        // }
+        update: (cache, { data }) => {
+            const newCategory = data.createCourseCategory.courseCategory;
+            const cachedCategory = cache.readQuery({
```
refactor to "cachedCourseTopics"

dhuangg commented on src/components/FeatureViews/AdminPortal/CreateTopicModal.js:
```diff
@@ -90,57 +95,44 @@ const GET_COURSE_TAGS = gql`
-        tagName: '',
-        tagDescription: '',
+const CreateTopicModal = ({ closeModal }) => {
+    const [courseTopicData, setCourseTopicData] = useState({
```
refactor "courseTopicData" to "courseTopicValue". Data = database, Value = state values

pswk1 committed:
refactor(manage course topics): rename variables

- Update variable names in mutation function of create topic modal from "category" to "topic"

pswk1 committed:
Update ManageCourseTopics.js

pswk1 committed:
Apply formatting changes

dhuangg reviewed (approved):
LOOKS GOOD TO MEEEEE
